hazelcast/src/main/java/com/hazelcast/map/impl/record/AbstractRecord.java

```diff
@@ -18,85 +18,140 @@
 
 import com.hazelcast.nio.serialization.Data;
 
+import static com.hazelcast.map.impl.record.RecordStatistics.EMPTY_STATS;
 
+/**
+ * @param <V> the type of the value of Record.
+ */
 @SuppressWarnings("VolatileLongOrDoubleField")
-abstract class AbstractRecord<V> extends AbstractBaseRecord<V> {
+public abstract class AbstractRecord<V> implements Record<V> {
 
     protected Data key;
+    protected long version;
+    /**
+     * evictionCriteriaNumber may be used for LRU or LFU eviction depending on configuration.
+     */
+    protected long evictionCriteriaNumber;
+    protected long ttl;
+    protected long lastAccessTime;
+    protected long lastUpdateTime;
+    protected long creationTime;
+
+    AbstractRecord() {
+        version = 0L;
+    }
 
-    public AbstractRecord(Data key) {
-        this.key = key;
+    @Override
+    public final long getVersion() {
+        return version;
     }
 
-    public AbstractRecord() {
+    @Override
+    public final void setVersion(long version) {
+        this.version = version;
     }
 
     @Override
-    public final Data getKey() {
-        return key;
+    public long getEvictionCriteriaNumber() {
+        return evictionCriteriaNumber;
     }
 
     @Override
-    public RecordStatistics getStatistics() {
-        return null;
+    public void setEvictionCriteriaNumber(long evictionCriteriaNumber) {
+        this.evictionCriteriaNumber = evictionCriteriaNumber;
     }
 
     @Override
-    public void setStatistics(RecordStatistics stats) {
+    public long getTtl() {
+        return ttl;
     }
 
     @Override
-    public void onAccess() {
+    public void setTtl(long ttl) {
+        this.ttl = ttl;
     }
 
     @Override
-    public void onStore() {
+    public long getLastAccessTime() {
+        return lastAccessTime;
     }
 
     @Override
-    public void onUpdate() {
-        version++;
+    public void setLastAccessTime(long lastAccessTime) {
+        this.lastAccessTime = lastAccessTime;
     }
 
     @Override
-    public Object getCachedValue() {
-        return Record.NOT_CACHED;
+    public long getLastUpdateTime() {
+        return lastUpdateTime;
     }
 
     @Override
-    public void setCachedValue(Object cachedValue) {
+    public void setLastUpdateTime(long lastUpdateTime) {
+        this.lastUpdateTime = lastUpdateTime;
+    }
 
+    @Override
+    public long getCreationTime() {
+        return creationTime;
+    }
+
+    @Override
+    public void setCreationTime(long creationTime) {
+        this.creationTime = creationTime;
     }
 
     @Override
     public long getCost() {
```

Comment:
It may not be the right calculation but this is not what i am trying to address with this PR and it should be the same calculation what we had before this PR, this PR does not touch it.